main.go

```diff
@@ -178,7 +180,10 @@ func main() { //nolint:funlen,maintidx
 				Field: fields.Set{"metadata.name": "default"}.AsSelector(),
 			},
 			// for prometheus and black-box deployment and ones we owns
-			&appsv1.Deployment{}: {Namespaces: deploymentCache},
+			&appsv1.Deployment{}: {
+				Namespaces: deploymentCache,
+				Label:      pkglables.Set{labels.ODH.Component(modelregistry.ModelRegSelector): "true"}.AsSelector(),
```

Comment:
not sure I get the rationale here, are we only expected to cache model registry's deployments ?

Reply:
you are right. need to move the selector intoe deploymentCache, otherwise, it will only get the modelreg deployment and skip all others